ui/app/components/raft-storage-overview.js

```diff
@@ -0,0 +1,67 @@
+import Component from '@ember/component';
+import { getOwner } from '@ember/application';
+import config from '../config/environment';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+  flashMessages: service(),
+  useServiceWorker: null,
+
+  async init() {
+    this._super(...arguments);
+    if (this.useServiceWorker === false) {
+      return;
+    }
+    if ('serviceWorker' in navigator) {
+      let worker = await navigator.serviceWorker.getRegistration(config.serviceWorkerScope);
+      if (worker) {
+        this.set('useServiceWorker', true);
+      }
+    }
+  },
+
+  actions: {
+    async removePeer(model) {
+      let { nodeId } = model;
+      try {
+        await model.destroyRecord();
+      } catch (e) {
+        let errString = e.errors ? e.errors.join(' ') : e.message || e;
+        this.flashMessages.danger(`There was an issue removing the peer ${nodeId}: ${errString}`);
+        return;
+      }
+      this.flashMessages.success(`Successfully removed the peer: ${nodeId}.`);
+    },
+
+    downloadViaServiceWorker() {
+      this.flashMessages.success('The snapshot download will begin shortly.');
```

Comment:
when i saw the name of this method i expected to see something like:

```
downloadViaServiceWorker() {
      this.flashMessages.success('The snapshot download will begin shortly.');
      this.initializeDownload() // something that actually kicks off the download
}
```

if this message merely sets a flash message, where does the download actually happen?  
is it happening under the hood because the `a` element that queues this action has `/v1/sys/storage/raft/snapshot` as it's `href`?

Reply:
Yeah, clicking the link links directly to the endpoint and its response has a `Content-Disposition` header to start a download (I linked the PRs that added that in the description). We can't _just_ link to it though because the endpoint needs a Vault token to be authenticated and there's no native way to do this with an anchor link. The Service Worker we added intercepts this request in a new thread, and communicates to our application thread in order to get the current Vault token, and then adds this the necessary header to the request, allowing the download to work with just the anchor tag.

Reply:
oh nice! thanks for the explanation. in that case could you drop a short comment here explaining that the service worker intercepts the request so we know where to look if we need to debug it later?